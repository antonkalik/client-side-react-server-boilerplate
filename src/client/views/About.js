import React, { useEffect } from 'react';
import { connect } from 'react-redux';

function About({ store }) {
  useEffect(() => {
    console.log({ store });
  }, [store]);

  return <div className="about">About page</div>;
}

const mapStateToProps = store => {
  return { store };
};

export default connect(
  mapStateToProps,
  null
)(About);
