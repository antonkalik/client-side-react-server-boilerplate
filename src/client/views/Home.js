import React, { useEffect, useState } from 'react';
import { Header, Content } from '../components';

let intervalId;

export default function Home() {
  const [isOpen, setOpen] = useState(false);
  const [count, setCount] = useState(0);

  useEffect(() => {
    if (isOpen) {
      intervalId = setInterval(() => {
        setCount(count + 1);
      }, 1000);
    }
    return () => {
      clearInterval(intervalId);
    };
  }, [isOpen, intervalId]);

  console.log(count);

  return (
    <div className="home">
      <Header />
      <Content />

      <button onClick={() => setOpen(!isOpen)}>Click me</button>
      <span> </span>
      <label>{isOpen ? 'On' : 'Off'}</label>
      <div>
        <h1>Timer from useEffect:</h1>
        {Array(count)
          .fill('*')
          .map((it, i) => (
            <p key={i}>{i}</p>
          ))}
      </div>
    </div>
  );
}
