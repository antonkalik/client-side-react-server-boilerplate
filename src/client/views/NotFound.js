import React from 'react';

export default function NotFound() {
  return <div className="not-found">Not Found page</div>;
}
