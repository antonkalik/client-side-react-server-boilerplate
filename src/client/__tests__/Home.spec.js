import React from 'react';
import { shallow } from 'enzyme';
import { Home } from '../views/Home';

describe('<Home /> test', () => {
  test('should render correctly', () => {
    const component = shallow(<Home />);
    expect(component.find('Header')).toHaveLength(1);
    expect(component.find('Content')).toHaveLength(1);
    expect(component.hasClass('home')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
