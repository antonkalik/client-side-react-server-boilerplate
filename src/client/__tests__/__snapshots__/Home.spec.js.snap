// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> test should render correctly 1`] = `ShallowWrapper {}`;
